hazelcast/src/main/java/com/hazelcast/map/impl/PartitioningStrategyFactory.java

```diff
@@ -47,17 +51,31 @@ public PartitioningStrategyFactory(ClassLoader configClassLoader) {
      * first attempts locating a {@link PartitioningStrategy} in {code config.getPartitioningStrategy()}. If this is {@code null},
      * then looks up its internal cache of partitioning strategies; if one has already been created for the given
      * {@code mapName}, it is returned, otherwise it is instantiated, cached and returned.
-     * @param mapName       Map for which this partitioning strategy is being created
-     * @param config        the partitioning strategy configuration
+     *
+     * @param mapName                      Map for which this partitioning strategy is being created
+     * @param config                       the partitioning strategy configuration
+     * @param partitioningAttributeConfigs
      * @return
      */
-    public PartitioningStrategy getPartitioningStrategy(String mapName, PartitioningStrategyConfig config) {
+    @SuppressWarnings("checkstyle:NestedIfDepth")
+    public PartitioningStrategy getPartitioningStrategy(
+            String mapName,
+            PartitioningStrategyConfig config,
+            final List<PartitioningAttributeConfig> partitioningAttributeConfigs
+    ) {
         PartitioningStrategy strategy = null;
         if (config != null) {
```

Comment:
Because of this line the strategy isn't used at all, if no strategy, but only attributes.